run-tests.php

```diff
@@ -2574,19 +2567,20 @@ function run_test(string $php, $file, array $env): string
             $wanted_re = $temp;
 
             // Stick to basics
-            $wanted_re = str_replace('%e', '\\' . DIRECTORY_SEPARATOR, $wanted_re);
-            $wanted_re = str_replace('%s', '[^\r\n]+', $wanted_re);
-            $wanted_re = str_replace('%S', '[^\r\n]*', $wanted_re);
-            $wanted_re = str_replace('%a', '.+', $wanted_re);
-            $wanted_re = str_replace('%A', '.*', $wanted_re);
-            $wanted_re = str_replace('%w', '\s*', $wanted_re);
-            $wanted_re = str_replace('%i', '[+-]?\d+', $wanted_re);
-            $wanted_re = str_replace('%d', '\d+', $wanted_re);
-            $wanted_re = str_replace('%x', '[0-9a-fA-F]+', $wanted_re);
-            $wanted_re = str_replace('%f', '[+-]?\.?\d+\.?\d*(?:[Ee][+-]?\d+)?', $wanted_re);
-            $wanted_re = str_replace('%c', '.', $wanted_re);
-            $wanted_re = str_replace('%0', '\x00', $wanted_re);
-            // %f allows two points "-.0.0" but that is the best *simple* expression
+            $wanted_re = strtr($wanted_re, [
+                '%e' => '\\' . DIRECTORY_SEPARATOR,
```

Comment:
```suggestion
                '%e' => preg_quote(DIRECTORY_SEPARATOR, '/'),
```

https://3v4l.org/72B5A

Reply:
Nice, thank you. I merged these changes to the existing commit (with co-author remark) to include this and the other change in `preg_match` call.